compiler/src/main/java/com/squareup/anvil/compiler/codegen/ksp/KspUtil.kt

```diff
@@ -365,7 +366,24 @@ internal fun KSType.contextualToTypeName(
   typeParamResolver: TypeParameterResolver = TypeParameterResolver.EMPTY,
 ): TypeName {
   checkErrorType(origin)
-  return toTypeName(typeParamResolver)
+  return try {
+    toTypeName(typeParamResolver)
+  } catch (e: Exception) {
```

Comment:
Under what circumstance does this exception get thrown but passes the checkErrorType() line above? I'm a little confused how this situation is happening as this seems to be reimplementing KSP's built-in error reporting 🤔

Reply:
In the one present in the test case, the parameter has a type with some generic that's not imported/could not be found (`List<Foo>`). I guess that only reports the `List` type is fine and doesn't check the `Foo` one until `toTypeName()` is used?

Reply:
I changed `checkErrorType` implementation so it checks the arguments too, which is what was missing. I also added some extra context in case the node was a parameter. Let me know if that's better.